lib/srv/sess.go

```diff
@@ -919,7 +925,9 @@ func (s *session) emitSessionJoinEvent(ctx *ServerContext) {
 		sessionJoinEvent.ConnectionMetadata.LocalAddr = ctx.ServerConn.LocalAddr().String()
 	}
 
-	// Emit session join event to Audit Log.
+	if err := s.recordEvent(ctx.srv.Context(), sessionJoinEvent); err != nil {
+		s.log.WithError(err).Warn("Failed to record session join event.")
+	}
```

Comment:
i am assuming the reason `s.recordEvent` is before `s.emitAuditEvent` because recordEvent does the setup?

Reply:
Yes exactly